### PR TITLE
feat: add-huggingface-featurizer

### PR DESCRIPTION
**Pull request by frillecode**
- [ ] I have battle-tested on Overtaci (RMAPPS1279)
- [ ] I have bumped the version as described in the [wiki](https://www.notion.so/Setting-up-Poetry-on-your-local-machine-76647d75e8fa4d9ba553cbc2a49946d9#dca1e0f2ce934a49acd1851aaf882d75)

## Notes for reviewers
Could not think of a better test for `test_add_hf_text_data()` for now since the output will depend on which model from HF one chooses to load. Might return and add more testing if I think of something great.

### CODE REVIEW

**Comment by frillecode** [comment]
Need to write a test for the `_chunk_text()` function

**Comment by MartinBernstorff** [comment]
Remember to have at least one commit titled "feat: X" to bump the minor version 👍

**Review by HLasse — COMMENTED** [review]
Great work, Frida - this will be super valuable! 
I just want to get @KennethEnevoldsen's thoughts on the main loop (lines 207-225 in `load_text.py`). Do you see any way we can parallelize this or speed things up? I'm unsure how `Pool` would behave when using a model, and batching has to handled in a clever way due to different sized inputs (varying chunk sizes)

File: src/psycop_feature_generation/loaders/raw/load_text.py [hunk 1]
```diff
@@ -107,14 +111,125 @@ def _tfidf_featurize(
...
+
+        with torch.no_grad():
+            model_output = model(**encoded_input)
+
+        embedding = _mean_pooling(model_output, encoded_input["attention_mask"])
+
+        if len(chunks) > 1:
+            list_of_embeddings.append(torch.mean(embedding, axis=0).numpy())
+        else:
+            list_of_embeddings.append(embedding.numpy()[0])
```
[the remark this answers is not in the document]
Comment (KennethEnevoldsen) [reply src/psycop_feature_generation/loaders/raw/load_text.py:225]: See fridas comment (down)

**Comment by frillecode** [comment]
Kenneth and Frida discussed:
1. Optimization
- How fast is it on the actual data?
- Consider feeding all texts to the model in the same batch (or e.g. 60 at a time)
2. Caching
- Need to be able to cache it (currently can't handle more than 1 column?) 
3. Document representation
- How do we best represent the text? Are the averaged embeddings meaningful? Doc similarity
- And how important is it that it is meaningful? depends on what we are using it for 
4. Chunking logic
- More neat workaround max_seq_length. Ideally, it should be done after tokenization (as discussed earlier also). Check danishfoundationmodels 
- Keep whole sentences in chunks

We decided (with Martin also) to merge the PR for now and wait with making the discussed changes until we are closer to having to use it.

**Pull request merged**